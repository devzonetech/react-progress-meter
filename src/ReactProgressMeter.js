import React from 'react';
import PropTypes from 'prop-types';
import "!style-loader!css-loader!./css/styles.css";
import "!style-loader!css-loader!./css/bar.styles.css";

class ReactProgressMeter extends React.Component {
    
    constructor() {
        super();
    }
    
    render() {
        
        var progressClasses = "bar bar-" + this.props.currentProgress + " " + this.props.color;
        var styleProps = {
            width: this.props.width,
        }
        
        return (
            <div style={styleProps} className={this.props.show ? "chart showProgress" : "chart hideProgress"}>
                <div className={progressClasses}>
                    <div className="face top">
                        <div className="growing-bar"></div>
                    </div>
                    <div className="face side-0">
                        <div className="growing-bar"></div>
                    </div>
                    <div className="face floor">
                        <div className="growing-bar">{this.props.showPercent ? this.props.currentProgress + '%' : '' }</div>
                    </div>
                    <div className="face side-a"></div>
                    <div className="face side-b"></div>
                    <div className="face side-1">
                        <div className="growing-bar"></div>
                    </div>
                </div>
            </div>);    
    }
}

ReactProgressMeter.propTypes = {
    currentProgress: PropTypes.number,
    show: PropTypes.bool,
    showPercent: PropTypes.bool,
    color: PropTypes.string,
    width: PropTypes.string,
};

ReactProgressMeter.defaultProps = {
    currentProgress: 75,
    show: true,
    showPercent: false,
    color: "cyan",
    width: "100%",
};

export default ReactProgressMeter;