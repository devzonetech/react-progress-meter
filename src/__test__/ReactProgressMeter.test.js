import React from 'react'
import renderer from 'react-test-renderer'
import ReactProgressMeter from '../ReactProgressMeter'

describe('ReactProgressMeter', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<ReactProgressMeter currentProgress={45} color="lime" />)
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})