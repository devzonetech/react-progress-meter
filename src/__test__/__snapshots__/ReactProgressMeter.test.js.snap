// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactProgressMeter snapshot renders 1`] = `
<div
  className="chart showProgress"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <div
    className="bar bar-45 lime"
  >
    <div
      className="face top"
    >
      <div
        className="growing-bar"
      />
    </div>
    <div
      className="face side-0"
    >
      <div
        className="growing-bar"
      />
    </div>
    <div
      className="face floor"
    >
      <div
        className="growing-bar"
      >
        
      </div>
    </div>
    <div
      className="face side-a"
    />
    <div
      className="face side-b"
    />
    <div
      className="face side-1"
    >
      <div
        className="growing-bar"
      />
    </div>
  </div>
</div>
`;
